spark/dl/src/main/scala/com/intel/analytics/bigdl/models/autoencoder/Train.scala

```diff
@@ -35,7 +36,8 @@ object toAutoencoderBatch {
 class toAutoencoderBatch[T] extends Transformer[MiniBatch[T], MiniBatch[T]] {
   override def apply(prev: Iterator[MiniBatch[T]]): Iterator[MiniBatch[T]] = {
     prev.map(batch => {
-      MiniBatch(batch.data, batch.data)
+      MiniBatch(batch.getInput().asInstanceOf[Tensor[T]],
+        batch.getInput().asInstanceOf[Tensor[T]])
```

Comment:
To avoid user application code change, we can add two deprecated method `data` and `label`, which return the input and target tensors respectively.